.golangci.yml

```diff
@@ -0,0 +1,153 @@
+# options for analysis running
+run:
+  # timeout for analysis, e.g. 30s, 5m, default is 1m
+  timeout: 1m
+
+output:
+  sort-results: true
+
+# Uncomment and add a path if needed to exclude
+# skip-dirs:
+#   - some/path
+# skip-files:
+#   - ".*\\.my\\.go$"
+#   - lib/bad.go
+
+# Find the whole list here https://golangci-lint.run/usage/linters/
+linters:
+  disable-all: true
+  enable:
+    - deadcode # finds unused code
+    - errcheck # checking for unchecked errors in go programs
+    - errorlint # errorlint is a linter for that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13.
+    # - goconst # finds repeated strings that could be replaced by a constant
```

Comment:
even in this small repo it complains a lot about each string value repeated twice. I don't think it's worth it. I looked at some code in beats and it's very common, I don't think we should enforce creating constants for everything but if the team votes differently I'll enable it, hence it's still commented in the config.

Reply:
Well, this is still a code smell and prevent typos error.

Reply:
I'd like to see more opinions from the rest of the team, I don't have a strong opinion against adding this linter.

Reply:
To be honest, I've seen codebases using both approaches and the ones without the constants were easier to read/understand because one did not have to be jumping back and forth between the constant usage and definitions to know their value/visible result.

@ph does have a point regarding typos. Again, not a strong opinion.

Reply:
I don't have a strong opinion on this, I'm probably fine with either option.

For some of the linters we have that are more cosmetic or controversial, we could consider enabling them in a second pass after we have the linter setup everywhere. 

It is easier to reason about the consequences of enabling a particular linter if we can see what problems it is finding in a diff.